File: day09/src/main/java/de/havox_design/aoc2017/day09/Day09.java
package de.havox_design.aoc2017.day09;

import de.havox_design.aoc2017.utils.DataReader;

import java.util.List;

public class Day09 {
    private final List<String> input;

    public Day09(String fileName) {
        input = readData(fileName);
    }

    public static long solvePart1(String fileName) {
        Day09 instance = new Day09(fileName);
        return instance.solvePart1();
    }

    public static long solvePart2(String fileName) {
        Day09 instance = new Day09(fileName);
        return instance.solvePart2();
    }

    public long solvePart1() {
        return 0L;
    }

    public long solvePart2() {return 0L;
    }

    private List<String> readData(String fileName) {
        return DataReader.readData(fileName, MainClass.class);
    }
}

File: day09/src/main/scala/de/havox_design/aoc2017/day09/Day09.scala
package de.havox_design.aoc2017.day09

object Day09 {
  def solvePart1(filename: String): Int =
    calculate(readData(filename).next().trim)
      .groupScore

  def solvePart2(filename: String): Int =
    calculate(readData(filename).next().trim)
      .garbageScore

  def main(args: Array[String]): Unit = {
    def dayFileName = "day09.txt"

    println("Solution for part1: " + solvePart1(dayFileName))
    println("Solution for part2: " + solvePart2(dayFileName))
  }

  private def calculate(input: String): State =
    val initial = State(1, 0, false, 0, false)

    input
      .foldLeft(initial)((state, next) => state
        .step(next))

  private def readData(filename: String): Iterator[String] =
    scala
      .io
      .Source
      .fromResource(filename)
      .getLines()
}
